vendor/bat-native-ads/src/bat/ads/internal/client/client.cc

```diff
@@ -356,46 +353,30 @@ bool Client::ToggleSavedAd(const AdContentInfo& ad_content) {
   return is_saved;
 }
 
-bool Client::GetSavedAdForCreativeInstanceId(
-    const std::string& creative_instance_id) {
-  for (const auto& element : client_->ads_shown_history) {
-    if (element.ad_content.creative_instance_id == creative_instance_id) {
-      return element.ad_content.is_saved;
-    }
-  }
-
-  return false;
-}
-
 bool Client::ToggleFlaggedAd(const AdContentInfo& ad_content) {
   DCHECK(is_initialized_);
 
   const bool is_flagged = !ad_content.is_flagged;
 
-  const auto iter =
-      std::find_if(client_->ad_preferences.flagged_ads.cbegin(),
-                   client_->ad_preferences.flagged_ads.cend(),
-                   [&ad_content](const FlaggedAdInfo& flagged_ad) {
-                     return flagged_ad.creative_instance_id ==
-                            ad_content.creative_instance_id;
-                   });
+  const auto iter = std::find_if(
+      client_->ad_preferences.flagged_ads.cbegin(),
+      client_->ad_preferences.flagged_ads.cend(),
+      [&ad_content](const FlaggedAdInfo& flagged_ad) {
+        return flagged_ad.creative_set_id == ad_content.creative_set_id;
+      });
 
   if (is_flagged) {
-    if (iter == client_->ad_preferences.flagged_ads.end()) {
-      FlaggedAdInfo flagged_ad;
-      flagged_ad.creative_instance_id = ad_content.creative_instance_id;
-      flagged_ad.creative_set_id = ad_content.creative_set_id;
-      client_->ad_preferences.flagged_ads.push_back(flagged_ad);
-    }
+    FlaggedAdInfo flagged_ad;
+    flagged_ad.creative_set_id = ad_content.creative_set_id;
+    client_->ad_preferences.flagged_ads.push_back(flagged_ad);
```

Comment:
We could have duplicate of flagged ad if it was flagged before. Is it intended?

Reply:
Cleaned up code, please check again however this was not possible as we toggle the flag and if unflagged remove the item from the array